source_test.go

```diff
@@ -199,6 +200,66 @@ func TestSourcePluginAdapter_Run(t *testing.T) {
 	<-runDone
 }
 
+func TestSourcePluginAdapter_Run_Stuck(t *testing.T) {
+	is := is.New(t)
+	ctrl := gomock.NewController(t)
+	src := NewMockSource(ctrl)
+
+	teardownTimeout = time.Second
+	stopTimeout = time.Second
+	defer func() {
+		teardownTimeout = time.Minute // reset
```

Comment:
this feels a bit risky, what if something happened and the defer function didn't run? is that a possibility?

Reply:
Maybe it's better to make the timeouts "partially configurable", i.e. make them fields in `sourcePluginAdapter`. That would make them more usable in tests, less risky, and that code can be re-used when we make it "completely configurable".